### ISSUE TITLE
Class 'Symfony\Component\Translation\Translator' not found in vendor/nesbot/carbon/src/Carbon/Translator.php on line 7

### ISSUE THREAD

**Issue by m144** [briannesbitt/Carbon#1371]
When running composer update, carbon was updated from 1.30.0 to 1.31.0 and then I received the following message:
PHP Fatal error:  Class 'Symfony\Component\Translation\Translator' not found in vendor/nesbot/carbon/src/Carbon/Translator.php on line 7
I'm running laravel 5.5.40 with php 7.0.30 and symfony/translation version 3.4 .
When I manually fixed the version to 1.30.0 in the composer.json file, everything worked fine.
I checked with a co-worker with the same version of laravel , php version 7.1.17 and symfony/translation version 4.1.1 and everything worked for him, even with the latest version of carbon.

**Comment by kylekatarnls** [comment]
Thanks a lot, can you try to set your version to `dev-master` (our next 1.31.1 version), it should test Translator is available before to set the language.

And I will try to support symfony/translation 3 for our version 1.32.

**Comment by m144** [comment]
Great, thanks. I tried setting the version to dev-master, and I received the following error from the laravel requirements:
Your requirements could not be resolved to an installable set of packages.
  Problem 1    - laravel/framework v5.5.40 requires nesbot/carbon ^1.24.1 -> satisfiable by nesbot/carbon[1.24.1, 1.24.2, 1.25.0, 1.26.0, 1.26.1, 1.26.2, 1.26.3, 1.26.4, 1.27.0, 1.28.0, 1.29.0, 1.29.1, 1.29.2, 1.30.0, 1.31.0] but these conflict with your requirements or minimum-stability.

**Comment by kylekatarnls** [comment]
Maybe you can "falsify" the stability: `"nesbot/carbon": "dev-master as 1.24.1"` or simply set temporarily your project minimum-stability to `dev` (but this last affect all the dependencies).

**Comment by m144** [comment]
Great, the alias (dev-master as 1.30) worked for me.
It's not necessarily a good idea for production environments though, since it might get 'broken' versions.
So for now I'll set the version as 1.30 and as soon as version 1.31.1 is available I will upgrade.
Thanks!

**Comment by kylekatarnls** [comment]
Thanks to confirm the compatibility of `dev-master`. Patch released.

**Comment by m144** [comment]
@kylekatarnls The same issue came back with the latest update from 1.31.1 to 1.32.0

**Comment by kylekatarnls** [comment]
The `Symfony\Component\Translation\Translator` exist in symfony/translation from version 2.6:
https://github.com/symfony/translation/blob/2.6/Translator.php
https://github.com/symfony/translation/blob/3.0/Translator.php
https://github.com/symfony/translation/blob/4.0/Translator.php

And in 3.4 (your version): https://github.com/symfony/translation/blob/3.4/Translator.php

It means the Translator class should just simply be autoloaded. Can I have your composer.json?

**Comment by kylekatarnls** [comment]
I just installed a Laravel project from scratch (with laravel/framework 5.5.40) then required symfony/translation 3.4.12.

It installed nesbot/carbon 1.32.0 and `Discovered Package: nesbot/carbon` went just fine. Next `composer update` are fine too. Then I'm able create a carbon change the app locale and get the diffForHumans well translated.

So I assume there is a conflict with something else or an override somewhere.

**Comment by eth001** [comment]
> My bad, I had a problem in my settings. It's all good, thanks!

what the f*** was the problem ? how did you solve it ?

**Comment by Messhias** [comment]
> > My bad, I had a problem in my settings. It's all good, thanks!
> 
> what the f*** was the problem ? how did you solve it ?

the problem still persists, it's just happened now, for some reason the composer updated post-installation of the internal library seems busted (again). 

We'll need to check in the PR's if something was uploaded that messed up.

**Comment by kylekatarnls** [comment]
No, it's a completely different topic, read the message it's not the same, what you have today is likely just the broken version of symfony 5.3.8: https://github.com/symfony/symfony/issues/43212

Just don't install it and wait for Symfony to fix it.

**Comment by DDSameera** [comment]
i run this command "composer clear-cache" then problem has been fixed

**Comment by JahanzaibHussain** [comment]
![image](https://user-images.githubusercontent.com/16865035/135078983-fa02fe84-b4c7-41a7-81fa-276865e27adf.png)

in my case **composer clear-cache** not working.

**Comment by DDSameera** [comment]
@JahanzaibHussain , I faced same scenario and i run `composer clear-cache` , then problem has been resolved. I suggest you to remove these cache by manually in your computer. please take backup before the deletion

`
C:\Users\XXXX\AppData\Local\Composer\repo
C:\Users\XXX\AppData\Local\Composer\files
C:\Users\XXXX\AppData\Local\Composer`

`D:\Lectures\Laravel\Practicals>composer clear-cache
Cache directory does not exist (cache-vcs-dir):
Clearing cache (cache-repo-dir): C:\Users\digix\AppData\Local\Composer\repo
Clearing cache (cache-files-dir): C:\Users\digix\AppData\Local\Composer\files
Clearing cache (cache-dir): C:\Users\digix\AppData\Local\Composer
All caches cleared.`
